internal/coreinternal/processor/filtermetric/filtermetric.go

```diff
@@ -25,8 +25,36 @@ type Matcher interface {
 // NewMatcher constructs a metric Matcher. If an 'expr' match type is specified,
 // returns an expr matcher, otherwise a name matcher.
 func NewMatcher(config *MatchProperties) (Matcher, error) {
+	if config == nil {
+		return nil, nil
+	}
+
 	if config.MatchType == Expr {
 		return newExprMatcher(config.Expressions)
 	}
 	return newNameMatcher(config)
 }
+
+// Filters have the ability to include and exclude metrics based on the metric's properties.
+// The default is to not skip. If include is defined, the metric must match or it will be skipped.
+// If include is not defined but exclude is, metric will be skipped if it matches exclude. Metric
+// is included if neither specified.
+func SkipMetric(include, exclude Matcher, metric pdata.Metric) bool {
+	if include != nil {
+		// A false (or an error) returned in this case means the metric should not be processed.
+		i, err := include.MatchMetric(metric)
+		if !i || err != nil {
+			return true
+		}
+	}
+
+	if exclude != nil {
+		// A true (or an error) returned in this case means the span should not be processed.
```

Comment:
I think we need consistent behavior in case of error. Looks like we will not skip metrics in case of error in include matcher, but skip it in case of exclude matcher error.

Probably we should skip the metric if an error happens with any matcher. Ideally we should log the error with debug level. To do that, `Logger` from the processor instance should be passed here .

Reply:
From my understanding the metric is always skipped if an error is hit. When an error is hit in include we skip, and when it's hit in exclude we skip.

I agree about logging though, so I'll add a log message when we're skipping due to error.